segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java

```diff
@@ -91,13 +91,19 @@ public void close() {
     public final static class ThreadPool implements AutoCloseable {
         private final OpStatsLogger queueSize;
         private final OpStatsLogger activeThreads;
+        private final OpStatsLogger storageQueueSize;
+        private final OpStatsLogger storageActiveThreads;
         private final ScheduledExecutorService executor;
+        private final ScheduledExecutorService storageExecutor;
         private final ScheduledFuture<?> reporter;
 
-        public ThreadPool(ScheduledExecutorService executor) {
+        public ThreadPool(ScheduledExecutorService executor, ScheduledExecutorService storageExecutor) {
             this.executor = Preconditions.checkNotNull(executor, "executor");
+            this.storageExecutor = Preconditions.checkNotNull(storageExecutor, "storageExecutor");
             this.queueSize = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_QUEUE_SIZE);
             this.activeThreads = STATS_LOGGER.createStats(MetricsNames.THREAD_POOL_ACTIVE_THREADS);
+            this.storageQueueSize = STATS_LOGGER.createStats(MetricsNames.STORAGE_THREAD_POOL_QUEUE_SIZE);
```

Comment:
You need to close these 2 metrics too.

Reply:
good catch. `this.storageQueueSize.close();` was missing